docs/en/reference/transactions.rst

```diff
@@ -48,27 +53,46 @@ constants:
     TransactionIsolationLevel::SERIALIZABLE
 
 The default transaction isolation level of a
-``Doctrine\DBAL\Connection`` is chosen by the underlying platform
-but it is always at least ``READ_COMMITTED``.
+``Doctrine\DBAL\Connection`` instance is chosen by the underlying
+platform but it is always at least ``READ_COMMITTED``.
+
+Emulated Transaction Nesting
+----------------------------
+
+Calling ``beginTransaction()`` while already in a transaction will
+result in two very different behaviors depending on whether transaction
+nesting with savepoints is enabled or not. In both cases though, there
+won't be an actual transaction inside a transaction, even if your RDBMS
+supports it. There is always only a single, real database transaction.
 
-Transaction Nesting
--------------------
+By default, transaction nesting at the SQL level with savepoints is
+disabled. The value for that setting can be set on a per-connection
```

Comment:
I'm not sure why this is the default by the way, if someone does read the docs and nests transaction, I think they would be less surprised when seeing a savepoint than when seeing a transaction. By the way, it was a long time ago but I think I remember seeing savepoint in my logs, but I don't remember configuring them. Maybe there is (was?) a default to `true` in some other layer (I've checked the DoctrineBundle and the symfony recipes, and that does not seem to be the case).

Reply:
I think from the perspective pf the API semantics, the default behavior is potentially harmful. If a given component starts and commits its transaction, it may expect that they are committed but they may be not depending on the configuration and the current nesting level. The same is about rollback. The more I think about how it works, the less sense it makes.

Reply:
Do I sense an incoming deprecation? DB2 does not support savepoints, so that would mean DB2 users would no longer have the option of having nested transactions, and this documentation block should be restored just for them:

> This also means that you cannot
successfully commit some changes in an outer transaction if an
inner transaction block fails and issues a rollback, even if this
would be the desired behavior (i.e. because the nested operation is
"optional" for the purpose of the outer transaction block). To
achieve that, you need to restructure your application logic so as
to avoid nesting transaction blocks. If this is not possible
because the nested transaction blocks are in a third-party API
you're out of luck.

Reply:
The default behavior should be definitely deprecated. If a platform doesn't support savepoints and there is an attempt to start a nested transaction, it should fail immediately.

The documentation block looks reasonable but it could use some cleanup in terms of language. Instead of focusing on the user ("you cannot [...] commit", "you're out of luck"), focus on the problems and the API.

Reply:
Will do!